Engine crashes when prefix caching hits new complete prefill in Metal paged unified path

## Summary

The Metal paged unified path crashes on long-context serving workloads when vLLM core prefix caching is enabled.

The root cause is an assertion in the unified prefill path:

```text
AssertionError: new complete prefill with start_pos > 0 not supported
(prefix caching not yet implemented in unified path)
```

This happens when vLLM schedules a new request with `num_computed_tokens > 0` due to prefix-cache hits.

## Repro

Serve:
```bash
GLOO_SOCKET_IFNAME=lo0 \
VLLM_METAL_USE_PAGED_ATTENTION=1 \
VLLM_METAL_MEMORY_FRACTION=0.3 \
vllm serve Qwen/Qwen3-0.6B \
  --max-model-len 2048
```

Benchmark:
```bash
vllm bench serve \
  --backend vllm \
  --model Qwen/Qwen3-0.6B \
  --endpoint /v1/completions \
  --dataset-name sonnet \
  --dataset-path /tmp/sonnet.txt \
  --sonnet-input-len 1024 \
  --sonnet-output-len 128 \
  --num-prompts 20 \
  --request-rate 0.5 \
  --max-concurrency 1
```

## Result with prefix caching enabled

```text
============ Serving Benchmark Result ============
Successful requests:                     0
Failed requests:                         20
Maximum request concurrency:             1
Request rate configured (RPS):           0.50
Benchmark duration (s):                  40.00
Total input tokens:                      0
Total generated tokens:                  0
Request throughput (req/s):              0.00
Output token throughput (tok/s):         0.00
Peak output token throughput (tok/s):    0.00
Peak concurrent requests:                0.00
Total token throughput (tok/s):          0.00
---------------Time to First Token----------------
Mean TTFT (ms):                          0.00
Median TTFT (ms):                        0.00
P99 TTFT (ms):                           0.00
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          0.00
Median TPOT (ms):                        0.00
P99 TPOT (ms):                           0.00
---------------Inter-token Latency----------------
Mean ITL (ms):                           0.00
Median ITL (ms):                         0.00
P99 ITL (ms):                            0.00
==================================================
```

The server crashes with:
```text
AssertionError: new complete prefill with start_pos > 0 not supported
(prefix caching not yet implemented in unified path)
```

## Control experiment: disable vLLM core prefix caching

Serve:
```bash
GLOO_SOCKET_IFNAME=lo0 \
VLLM_METAL_USE_PAGED_ATTENTION=1 \
VLLM_METAL_MEMORY_FRACTION=0.3 \
vllm serve Qwen/Qwen3-0.6B \
  --max-model-len 2048 \
  --no-enable-prefix-caching
```

Same benchmark result:

```text
============ Serving Benchmark Result ============
Successful requests:                     20
Failed requests:                         0
Maximum request concurrency:             1
Request rate configured (RPS):           0.50
Benchmark duration (s):                  70.61
Total input tokens:                      20233
Total generated tokens:                  2560
Request throughput (req/s):              0.28
Output token throughput (tok/s):         36.26
Peak output token throughput (tok/s):    46.00
Peak concurrent requests:                2.00
Total token throughput (tok/s):          322.81
---------------Time to First Token----------------
Mean TTFT (ms):                          215.64
Median TTFT (ms):                        204.21
P99 TTFT (ms):                           265.38
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          25.56
Median TPOT (ms):                        23.89
P99 TPOT (ms):                           32.26
---------------Inter-token Latency----------------
Mean ITL (ms):                           25.56
Median ITL (ms):                         23.00
P99 ITL (ms):                            43.87
==================================================
```

## Expected behavior

The Metal paged unified path should either:
- support `new request + start_pos > 0` / `num_computed_tokens > 0`, or
- explicitly disable incompatible vLLM core prefix caching when this path is active.

--- a/vllm_metal/platform.py
+++ b/vllm_metal/platform.py
@@ -232,6 +232,15 @@ class MetalPlatform(Platform):
             scheduler_config.enable_chunked_prefill = False
             logger.info("Metal: disabled chunked prefill")
 
+        if config.use_paged_attention and getattr(
+            cache_config, "enable_prefix_caching", False
+        ):
+            # The unified paged path does not yet safely support vLLM core
+            # prefix-cache hits for new requests. Disable the feature at the
+            # platform layer until that path is fully supported.
+            cache_config.enable_prefix_caching = False
+            logger.info("Metal: disabled prefix caching")
+
         # Configure cache
         if cache_config.block_size is None:
             cache_config.block_size = config.block_size

